File: targets/dotnet/aspnet-mvc/suite/ParametersTests.cs
using RequestBench.Suite;

namespace RequestBench.AspNetMvc.Suite;

/// <summary>
/// parameters: route capture, at zero, one and two segments.
///
/// The captured values do not reach the answer. The payload is the shared one, so what these
/// hold is that the route matched at all: a target whose two-segment pattern is wrong answers
/// 404 and the floor says so on the status line before it ever looks at a body.
/// </summary>
public sealed class ParametersTests(TargetApp app) : IClassFixture<TargetApp>
{
    // rb:test parameters.static
    [Fact]
    public async Task A_route_with_nothing_to_capture_matches()
    {
        Ask ask = Plan.For("parameters.static");

        using HttpResponseMessage response = await app.Send(ask);

        await Floor.AssertAsync(response, ask);
    }

    // rb:test parameters.one
    [Fact]
    public async Task One_captured_segment_matches()
    {
        Ask ask = Plan.For("parameters.one");

        using HttpResponseMessage response = await app.Send(ask);

        await Floor.AssertAsync(response, ask);
    }

    // rb:test parameters.two
    [Fact]
    public async Task Two_captured_segments_match()
    {
        Ask ask = Plan.For("parameters.two");

        using HttpResponseMessage response = await app.Send(ask);

        await Floor.AssertAsync(response, ask);
    }
}

File: targets/java/quarkus/src/test/java/rb/quarkus/ParametersTests.java
package rb.quarkus;

import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Test;

/**
 * parameters: route capture, at zero, one and two segments.
 *
 * <p>The captured values do not reach the answer. The payload is the shared one, so what these
 * hold is that the route matched at all: a target whose two-segment pattern is wrong answers
 * 404 and the floor says so on the status line before it ever looks at a body.
 */
@QuarkusTest
class ParametersTests extends QuarkusSuite {

  // rb:test parameters.static
  @Test
  void a_route_with_nothing_to_capture_matches() throws Exception {
    Planned.Ask a = Planned.ask("parameters.static");

    Floor.Answer answer = send(a);

    Floor.check(a, answer);
  }

  // rb:test parameters.one
  @Test
  void one_captured_segment_matches() throws Exception {
    Planned.Ask a = Planned.ask("parameters.one");

    Floor.Answer answer = send(a);

    Floor.check(a, answer);
  }

  // rb:test parameters.two
  @Test
  void two_captured_segments_match() throws Exception {
    Planned.Ask a = Planned.ask("parameters.two");

    Floor.Answer answer = send(a);

    Floor.check(a, answer);
  }
}
